pyphare/pyphare/pharesee/hierarchy.py

```diff
@@ -173,6 +174,36 @@ def plot(self):
 
 
 
+    def plot(self, qty, xlim=None, ylim=None, title=None,filename=None):
+        import matplotlib.pyplot as plt
+
+        times = np.sort(np.asarray(list(self.time_hier.keys())))
+        fig, ax = plt.subplots(figsize=(10, 6))
+        t = times[0]
+        for il, level in self.levels(t).items():
+            patches = level.patches
+            if il == 0:
+                marker = "+"
+                alpha = 1
+                ls = '-'
+            else:
+                marker = "o"
+                alpha = 0.4
+                ls = 'none'
+            for ip, patch in enumerate(patches):
+                val = patch.patch_datas[qty].dataset[:]
+                x_val = patch.patch_datas[qty].x
+                ax.plot(x_val, val, label=r"Level {} patch {}".format(il, ip), marker=marker, alpha=alpha, ls=ls)
+                if ylim is not None:
+                    ax.set_ylim(ylim)
+        ax.legend(ncol=4)
+        ax.set_title(title)
+
+        if filename is not None:
+            fig.savefig(filename)
```

Comment:
should figs be cleared after saving or something?
I think it's a static instance, so they can conflict if done in succession.

Reply:
I'm not following. If done in succession it's the caller's responsibility to change filename accordingly, no ?

Reply:
dunno, hence why I ask

Reply:
ok cool. Since `filename` is an argument I say it's the user responsibility to check it's not dumb to call the function with that name.